src/index.js

```diff
@@ -228,44 +243,76 @@ function applyMacros({
   return result
 }
 
-// eslint-disable-next-line consistent-return
-function getConfig(macro, filename, source) {
-  if (macro.options.configName) {
-    try {
-      // lazy-loading it here to avoid perf issues of loading it up front.
-      // No I did not measure. Yes I'm a bad person.
-      // FWIW, this thing told me that cosmiconfig is 227.1 kb of minified JS
-      // so that's probably significant... https://bundlephobia.com/result?p=cosmiconfig@3.1.0
-      // Note that cosmiconfig will cache the babel-plugin-macros config 👍
-      const explorer = require('cosmiconfig')('babel-plugin-macros', {
-        searchPlaces: [
-          'package.json',
-          `.babel-plugin-macrosrc`,
-          `.babel-plugin-macrosrc.json`,
-          `.babel-plugin-macrosrc.yaml`,
-          `.babel-plugin-macrosrc.yml`,
-          `.babel-plugin-macrosrc.js`,
-          `babel-plugin-macros.config.js`,
-        ],
-        packageProp: 'babelMacros',
-        sync: true,
-      })
-      const loaded = explorer.searchSync(filename)
-      if (loaded) {
-        return loaded.config[macro.options.configName]
+function getConfigFromFile(configName, filename) {
+  try {
+    const loaded = configExplorer.searchSync(filename)
+
+    if (loaded) {
+      return {
+        options: loaded.config[configName],
+        path: loaded.filepath,
       }
-    } catch (error) {
+    }
+  } catch (e) {
+    return {error: e}
+  }
+  return {}
+}
+
+function getConfigFromOptions(configName, options) {
+  if (Object.prototype.hasOwnProperty.call(options, configName)) {
```

Comment:
Any reason this can't just be `options.hasOwnProperty(configName)`?

Reply:
Up to you. It's not a wholly safe assumption -- it is violated by `options = Object.create(null)`. If that's acceptable to you I'll change it.

Reply:
I've literally never seen anyone create an object like that in real code :)

Reply:
I got burned by it writing the `entries` method for iter-tools. Usually the only such objects you'll encounter are internal data structures though, never configurations. I'll change it. Any thoughts on the rest of the changes?